common/network-common/src/main/java/org/apache/spark/network/util/RocksDBProvider.java

```diff
@@ -100,7 +100,11 @@ public static RocksDB initRockDB(File dbFile, StoreVersion version, ObjectMapper
           // is unusable
           checkVersion(tmpDb, version, mapper);
         } catch (RocksDBException e) {
+          tmpDb.close();
           throw new IOException(e.getMessage(), e);
+        } catch (IOException ioe) {
+          tmpDb.close();
+          throw ioe;
```

Comment:
https://github.com/apache/spark/blob/83434af22284482d4e805d8be625f733b074928b/common/network-common/src/main/java/org/apache/spark/network/util/RocksDBProvider.java#L151-L152

For RocksDB, it may throw either RocksDBException or IOException